core/src/test/java/integration/kafka/api/PlaintextConsumerDupTest.java

```diff
@@ -0,0 +1,358 @@
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.common.record.TimestampType;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.serialization.ByteArraySerializer;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.common.utils.Utils;
+import static org.apache.kafka.test.TestUtils.DEFAULT_MAX_WAIT_MS;
+
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import scala.Function0;
+import scala.collection.JavaConverters;
+import scala.collection.mutable.ArrayBuffer;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Properties;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@Timeout(600)
+@SuppressWarnings("deprecation") // Added for Scala 2.12 compatibility for usages of JavaConverters
+public class PlaintextConsumerDupTest extends BaseConsumerTest {
+
+    @Test
+    public void testHeaders() {
+        int numRecords = 1;
+        ProducerRecord<byte[], byte[]> record = new ProducerRecord<>(tp().topic(), tp().partition(), null, "key".getBytes(), "value".getBytes());
+
+        record.headers().add("headerKey", "headerValue".getBytes());
+
+        KafkaProducer<byte[], byte[]> producer = createProducer(new ByteArraySerializer(), new ByteArraySerializer(), new Properties());
+        producer.send(record);
+
+        Consumer<byte[], byte[]> consumer = createConsumer(new ByteArrayDeserializer(), new ByteArrayDeserializer(),
+                new Properties(), JavaConverters.asScalaBuffer(Collections.<String>emptyList()).toList());
+        assertEquals(0, consumer.assignment().size());
+        consumer.assign(Collections.singleton(tp()));
+        assertEquals(1, consumer.assignment().size());
+
+        consumer.seek(tp(), 0);
+        ArrayBuffer<ConsumerRecord<byte[], byte[]>> records = consumeRecords(consumer, numRecords, Integer.MAX_VALUE);
+
+        assertEquals(numRecords, records.size());
+
+        for (Iterator<ConsumerRecord<byte[], byte[]>> iter = JavaConverters.asJavaIterator(records.toIterator()); iter.hasNext();) {
+            ConsumerRecord<byte[], byte[]> consumerRecord = iter.next();
+            Header header = consumerRecord.headers().lastHeader("headerKey");
+            if (header != null)
+                assertEquals("headerValue", new String(header.value()));
+        }
+    }
+
+    static class SerializerImpl implements Serializer<byte[]> {
```

Comment:
May be we can move serializer from `PlaintextConsumerTest` to `BaseConsumerTest` itself as like we have defined `TestClusterResourceListenerSerializer` in `BaseConsumerTest` rather duplicating the code.

Reply:
ack